api/src/main/java/org/openmrs/api/db/hibernate/search/LuceneQuery.java

```diff
@@ -249,12 +251,41 @@ protected FullTextSession getFullTextSession() {
 			}
 			termsFilter = new TermsFilter(terms);
 		}
-		
+
 		if (termsFilter != null) {
 			buildQuery();
 			fullTextQuery.setFilter(termsFilter);
 		}
-		
+
+		return this;
+	}
+
+	public LuceneQuery<T> skipSame(List<Object> baseList, String field){
```

Comment:
I'd rather we changed the signature to be called like this: query1.skipSame(fieldOfQuery1, new SkipSame(query2, fieldOfQuery2)) i.e. LuceneQuery<T> skipSame(String field, SkipSame... skipSame);